pulser/simulation/simulation.py

```diff
@@ -205,45 +213,87 @@ def _build_basis_and_op_matrices(self) -> None:
                 self.basis[proj[0]] * self.basis[proj[1]].dag()
             )
 
-    def _build_operator(self, op_id: str, *qubit_ids: Union[str, int],
-                        global_op: bool = False) -> qutip.Qobj:
+    def _build_operator(self, operations: dict,
+                        ) -> qutip.Qobj:
         """Create qutip.Qobj with nontrivial action at *qubit_ids."""
-        if global_op:
-            return sum(self._build_operator(op_id, q_id)
-                       for q_id in self._qdict)
-        if len(set(qubit_ids)) < len(qubit_ids):
-            raise ValueError("Duplicate atom ids in argument list.")
-        # List of identity operators, except for op_id where requested:
-        op_list = [self.op_matrix[op_id]
-                   if j in map(self._qid_index.get, qubit_ids)
-                   else self.op_matrix['I'] for j in range(self._size)]
+        op_list = [self.op_matrix['I'] for j in range(self._size)]
+        for operator, qubits in operations.items():
```

Comment:
I put this comment on the previous PR: Please generalize to keys being `Qobj`s not just id strings and values being either ids (`'atomN'`) or directly indexes (`3, 5`).

Reply:
I am not sure that you can make a key of a Qobj

Reply:
The function that would work in the general case would be one that takes a list of `(local_operator, position)` tuples and assigns the corresponding local operators from it, for example:

```python
my_sim.build_operator([(qutip.basis(2,1).proj(), 'atom1'), ('sigma_gr', 3), (my_custom_qobj, 2)])
```

Then the code reads the couples one by one and assigns the nontrivial local operators to the initial list of identity matrics.